US-EPA-CAMD/easey-camd-services
# Feature/6425 configuration manager

## Related Issues

- [#6425](https://github.com/US-EPA-CAMD/easey-ui/issues/6425)

## Main Changes:

- Added a check to the submission queueing process to ensure that all inactive monitoring plans for a facility have been submitted before attempting to submit any active monitoring plans.
  - For now, I kept this as simple as possible: the check is run against all items in the request payload, and if the check fails for any of them, it will fail with an error without queueing anything. In the future, we can consider making this "smarter", e.g. allowing inactive and active plans in a single request and letting the back-end sort them out.
 
## Steps to Test:

1. Use the configuration manager to create a new MP. For my testing, I used ORIS 63259 and created a new Stack/Pipe **CP1** active 11/01/2024, and two new UnitStackConfiguration records **DEPC2-CP1** and **DEPC3-CP1**, both active 11/01/2024.
2. Evaluate the newly active plan (**DEPC2, DEPC3, CP1**) and the newly ended plans (**DEPC2** and **DEPC3**). The new plan will evaluate with critical errors, but that can be ignored.
3. In the _Submit_ module, make sure the "Include Files with Critical Errors" radio input is selected, then search for the **DEPC2, DEPC3, CP1** plan. Select the plan for submittal and confirm. Verify a clear error message is displayed.
4. In the same module, search for and select the **DEPC2** and **DEPC3** plans. Submit the MPs, and confirm a success message is displayed.
5. Repeat step 3, but a success message should now show instead.
